liblouis/compileTranslationTable.c

```diff
@@ -2707,12 +2889,101 @@ compileRule(FileInfo *file, TranslationTableHeader **table,
 		}
 		return putCharDotsMapping(
 				file, ruleChars.chars[0], ruleDots.chars[0], displayTable);
+	case CTO_None: {
+		// check if token is a macro name
+		if (inScopeMacros) {
+			const MacroList *macros = *inScopeMacros;
+			while (macros) {
+				const Macro *m = macros->head;
+				if (token.length == strlen(m->name) &&
+						eqasc2uni((unsigned char *)m->name, token.chars, token.length)) {
+					if (!inScopeMacros) {
+						compileError(file, "Calling macros only allowed in table files.");
+						return 0;
+					}
+					FileInfo tmpFile;
+					memset(&tmpFile, 0, sizeof(tmpFile));
+					tmpFile.fileName = file->fileName;
+					tmpFile.lineNumber = file->lineNumber;
+					tmpFile.encoding = noEncoding;
+					tmpFile.status = 0;
+					tmpFile.linepos = 0;
+					tmpFile.linelen = 0;
+					int argument_count = 0;
+					CharsString *arguments =
+							malloc(m->argument_count * sizeof(CharsString));
+					while (argument_count < m->argument_count) {
+						if (getToken(file, &token, "macro argument"))
+							arguments[argument_count++] = token;
+						else
+							break;
+					}
+					if (argument_count < m->argument_count) {
+						compileError(file, "Expected %d arguments", m->argument_count);
+						return 0;
+					}
+					int i = 0;
```

Comment:
What if too many arguments are given? This seems to be silently ignored.

Reply:
Yes, just like with all other rules. Superfluous argument are comments.